daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -44,6 +49,23 @@ abstract class SequenceParserBase(
   override lazy val childProcessors = childParsers
 }
 
+abstract class UnorderedSequenceParserBase(
+  srd: SequenceRuntimeData,
+  choiceParser: ChoiceParser,
+  childParsersArg: Vector[Parser])
+  extends SequenceParserBase(srd, childParsersArg) {
+
+  override protected def parse(pstate: PState) = {
+
+    do {
+      choiceParser.parse(pstate)
+    } while (pstate.processorStatus eq Success)
+
+    pstate.setSuccess()
```

Comment:
I have added validation code for min/max occurs, although it ended up being more complex than originally expected.  Given that arrays within unordered sequences do not need to be contiguous, we were creating multiple DIArrays for the same element.  I was originally going to prevent the creation of multiple arrays, but that had issues if we backtracked.  So, in order to get a proper count we had to combine the arrays.